routers/admin.py

```diff
@@ -278,3 +466,193 @@ def resend_verification_email(user_id: Annotated[str, UserIdParam],
+            admin_record=admin_record,
+            db_session=db_session,
+        )
+    else:
+        _revoke_group_membership(
+            user_id=user_id,
+            group_id=group.value,
+            reason=payload.reason,
+            admin_record=admin_record,
+            db_session=db_session,
+        )
+
+    return _membership_response()
+
+
+@router.post("/users/{user_id}/platforms/{platform_id}/approve")
+def approve_platform_membership(user_id: Annotated[str, UserIdParam],
+                                platform_id: Annotated[str, ServiceIdParam],
+                                client: Annotated[Auth0Client, Depends(get_auth0_client)],
+                                approving_user: Annotated[SessionUser, Depends(get_current_user)],
+                                db_session: Annotated[Session, Depends(get_db_session)]):
+    platform = _parse_platform_or_404(platform_id)
+    admin_record = _get_or_create_db_user(
+        user_id=approving_user.access_token.sub,
+        client=client,
+        db_session=db_session,
+    )
+    _approve_platform_membership(
+        user_id=user_id,
+        platform=platform,
+        admin_record=admin_record,
+        db_session=db_session,
+    )
+    return _membership_response()
+
+
+@router.post("/users/{user_id}/platforms/{platform_id}/revoke")
+def revoke_platform_membership(user_id: Annotated[str, UserIdParam],
+                               platform_id: Annotated[str, ServiceIdParam],
+                               payload: RevokeServiceRequest,
+                               client: Annotated[Auth0Client, Depends(get_auth0_client)],
+                               revoking_user: Annotated[SessionUser, Depends(get_current_user)],
+                               db_session: Annotated[Session, Depends(get_db_session)]):
+    platform = _parse_platform_or_404(platform_id)
+    admin_record = _get_or_create_db_user(
+        user_id=revoking_user.access_token.sub,
+        client=client,
+        db_session=db_session,
+    )
+    _revoke_platform_membership(
+        user_id=user_id,
+        platform=platform,
+        reason=payload.reason,
+        admin_record=admin_record,
+        db_session=db_session,
+    )
+    return _membership_response()
+
+
+@router.post("/users/{user_id}/groups/{group_id}/approve")
+def approve_group_membership(user_id: Annotated[str, UserIdParam],
+                             group_id: Annotated[str, ServiceIdParam],
+                             client: Annotated[Auth0Client, Depends(get_auth0_client)],
+                             approving_user: Annotated[SessionUser, Depends(get_current_user)],
+                             db_session: Annotated[Session, Depends(get_db_session)]):
+    group = _parse_group_or_404(group_id)
+    admin_record = _get_or_create_db_user(
+        user_id=approving_user.access_token.sub,
+        client=client,
+        db_session=db_session,
+    )
+    _approve_group_membership(
```

Comment:
Need to check the admin's roles against the `BiocommonsUser.admin_roles` to confirm they have permission to approve/revoke for the specific group